core_api/src/dependencies.py

```diff
@@ -72,48 +75,91 @@ class ESQuery(TypedDict):
     user_uuid: UUID
 
 
+class ESParams(TypedDict):
+    size: int
+    num_candidates: int
+    match_boost: float
+    knn_boost: float
+    similarity_threshold: float
+
+
 @lru_cache(1)
 def get_es_retriever(
     env: Annotated[Settings, Depends(get_env)], es: Annotated[Elasticsearch, Depends(get_elasticsearch_client)]
-) -> ElasticsearchRetriever:
+) -> BaseRetriever:
     """Creates an Elasticsearch retriever runnable.
 
     Runnable takes input of a dict keyed to question, file_uuids and user_uuid.
 
     Runnable returns a list of Chunks.
     """
 
-    def es_query(query: ESQuery) -> dict[str, Any]:
+    def es_query(query: ESQuery, params: ESParams) -> dict[str, Any]:
         vector = get_embedding_model(env).embed_query(query["question"])
 
-        knn_filter = [{"term": {"creator_user_uuid.keyword": str(query["user_uuid"])}}]
+        query_filter = [{"term": {"creator_user_uuid.keyword": str(query["user_uuid"])}}]
 
         if len(query["file_uuids"]) != 0:
-            knn_filter.append({"terms": {"parent_file_uuid.keyword": [str(uuid) for uuid in query["file_uuids"]]}})
+            query_filter.append({"terms": {"parent_file_uuid.keyword": [str(uuid) for uuid in query["file_uuids"]]}})
 
         return {
-            "size": env.ai.rag_k,
+            "size": params["size"],
```

Comment:
```suggestion
            "size": env.ai.rag_k or params["size"],
```
or i f we arent using env.ai we should remove it (but i think its probably a good idea)